wordpress-mobile/WordPress-Android
# [Site Creation] Step 2: Remove site info site creation step

This PR removes site info site creation step screen (Step 2 of 3) as part of #11420 and #11421.

P.S. Review after https://github.com/wordpress-mobile/WordPress-Android/pull/11437 is merged to the Master PR.

**To test:**

- Go to My site
- Click Switch Site
- Click "+" button to launch Add new site
- Click Create WordPress.com site on Add new site
- Choose a segment Blog/Business/Professional/Blank Canvas to open Step 1
- Verify
    - Site Info site creation step is not displayed
    - Domains step is displayed as Step 1 of 1
    - Auto domain search is not triggered on domains step as site title is not received from previous step
    - User initiated domain search functions properly

![site-info-step-removed](https://user-images.githubusercontent.com/1405144/76615187-89f77200-6547-11ea-84b6-909af4ef2bda.gif)


PR submission checklist:
